server/types/socket-events.d.ts

```diff
@@ -107,7 +107,7 @@ interface ServerToClientEvents {
 		token: string;
 	}) => void;
 
-	"search:results": (response: {results: ClientMessage[]}) => void;
+	"search:results": (response: SearchResponse) => void;
```

Comment:
I don't see why you'd already drop the type information here, we are still on the server side so having more info is good no?